contracts/interchain-token/tests/test.rs

```diff
@@ -4,55 +4,515 @@ extern crate std;
 use axelar_soroban_std::{
     assert_invoke_auth_err, assert_invoke_auth_ok, assert_last_emitted_event,
 };
-use interchain_token::contract::InterchainToken;
-use interchain_token::InterchainTokenClient;
-use soroban_sdk::testutils::{Address as _, MockAuth, MockAuthInvoke};
-use soroban_sdk::{Address, Env, Symbol};
+use interchain_token::{contract::InterchainToken, InterchainTokenClient};
+use soroban_sdk::testutils::{MockAuth, MockAuthInvoke};
+use soroban_sdk::{
+    symbol_short,
+    testutils::{Address as _, AuthorizedFunction, AuthorizedInvocation, BytesN as _},
+    Address, BytesN, Env, IntoVal as _, Symbol,
+};
+use soroban_token_sdk::metadata::TokenMetadata;
+
+fn create_token<'a>(env: &Env, owner: &Address, minter: &Address) -> InterchainTokenClient<'a> {
+    let interchain_token_service = Address::generate(&env);
+    let token_id: BytesN<32> = BytesN::<32>::random(&env);
+    let token_meta_data = TokenMetadata {
+        decimal: 6,
+        name: "name".into_val(env),
+        symbol: "symbol".into_val(env),
+    };
+
+    let contract_id = env.register(
+        InterchainToken,
+        (
+            owner,
+            minter,
+            &interchain_token_service,
+            &token_id,
+            token_meta_data,
+        ),
+    );
+
+    InterchainTokenClient::new(env, &contract_id)
+}
+
+#[test]
+fn test() {
+    let env = Env::default();
+
+    let owner1 = Address::generate(&env);
+    let owner2 = Address::generate(&env);
+    let minter1 = Address::generate(&env);
+    let user1 = Address::generate(&env);
+    let user2 = Address::generate(&env);
+    let user3 = Address::generate(&env);
+
+    let token = create_token(&env, &owner1, &minter1);
+
+    assert_invoke_auth_ok!(minter1, token.try_mint(&minter1, &user1, &1000_i128));
+    assert_eq!(
```

Comment:
this is pretty ugly, and is causing duplication. Modify `assert_invoke_auth_ok` to do this check, it has all the necessary args already?

Reply:
agreed. it's duplicated, already `mock_auths` is mocking necessary auth invocations. updated.